### PR TITLE
Introduce contents.json

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/compiler/lock.cc [hunk 1]
```diff
@@ -935,20 +986,24 @@ PackageLock PackageLock::read(const std::string& lock_file_path,
...
   }
 
+  Map<std::string, Map<std::string, std::string>> mappings;
+  fill_package_mappings(&mappings, package_dirs, fs);
+
   Map<std::string, std::string> path_to_package;
   for (auto package_id : lock_content.packages.keys()) {
     auto entry = lock_content.packages.at(package_id);
 
     // For simplicity we are using a loop from -1 to length, so we don't
```
Comment (kasperl) [inline src/compiler/lock.cc:996]:
```suggestion
    // For simplicity we are using a loop from -2 to length, so we don't
```

Comment (kasperl) [reply src/compiler/lock.cc:996]: Would it be terrible to factor this out and introduce a helper that you can call twice before the loop?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/compiler/lock.cc (hunk 1 above)
Comment (floitsch) [reply src/compiler/lock.cc]: https://github.com/toitlang/toit/pull/2545

**Pull request merged**